File: JS_Advanced_October_2016/Exercise/07. Modules/01. People Modules/employee.js
export class Employee{
    constructor(name, age){
        if (new.target === Employee){
            throw new Error("Cannot instantiate directly.");
        }
        this.name = name;
        this.age = Number(age);
        this.salary = 0;
        this.tasks = [];
    }

    work(){
        let currentTask = this.tasks.shift();
        console.log(`${this.name}${currentTask}`);
        this.tasks.push(currentTask);
    }

    collectSalary(){
        console.log(`${this.name} received ${this.getSalary()} this month.`);
    }

    getSalary(){
        return this.salary;
    }
}

File: Java_Fundamentals_January_2017/OOP_Basics/Exersices/01. Defining Classes/src/CompanyRooster/Employee.java
package CompanyRooster;

public class Employee {
    public String name;
    public Double salary;
    public String position;
    public String department;

    public Double getSalary() {
        return salary;
    }

    public String getDepartment() {
        return department;
    }

    public String email;
    public Integer age;

    public Employee (String name, Double salary, String position, String department, String email, Integer age){
        this.name = name;
        this.salary = salary;
        this.position = position;
        this.department = department;
        this.email = email;
        this.age = age;
    }

    public Employee(String name, Double salary, String position, String department, String email) {
        this (name, salary, position, department, email, -1);
    }

    public Employee(String name, Double salary, String position, String department) {
        this (name, salary, position, department, "n/a", -1);
    }


}
